Sources/SpeziAccount/Views/SignupForm.swift

```diff
@@ -0,0 +1,139 @@
+//
+// This source file is part of the Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import OrderedCollections
+import SpeziViews
+import SwiftUI
+
+
+/// A generalized signup form used with arbitrary ``AccountService`` implementations.
+///
+/// A `Form` that collects all configured account values (a ``AccountValueConfiguration`` supplied to ``AccountConfiguration``)
+/// split into `Section`s according the their ``AccountKeyCategory`` (see ``AccountKey/category``).
+public struct SignupForm<Service: AccountService, Header: View>: View {
```

Comment:
1. It was intentional to not provide a repeat-password field for the signup view. I think, especially in the age of password manages, this is nowadays pretty common to simply the signup process. Similarly, we also don't ask for a repeat email field. If a user really messed up their password (and didn't save it in the password manager), they can always use a password reset link. On the other side, there was no strong reasoning for including a "repeat" password field in the change-password view. There was a tendency to have something that is more than a 1-step process. I'm happy to remove the repeat password field in the password change sheet.
2. Having the NameFields in non-bold is primarily caused by them NameFields being not part of SpeziAccount but provided through SpeziViews and unable to influence their visuals.
  In regard to the formatting, I would strongly argue to move away from the **bold** formatting for a more clean UI.
  In regard to the labels: for iOS-based UI elements it is common to render labels as part of the prompt (see [Text field prompts](https://developer.apple.com/documentation/swiftui/textfield#Text-field-prompts)). This saves the most screen space and allows the user to view all of the text input. However, this is only sensible in cases where the user can unambiguously identify the contents of the text field based on its input. For example, due to the "Credentials" section header and the text field type (e.g. secure fields showing dots), the user can easily identity that the first field must be the userId/e-mail and the second is the password field, even if there is text already entered in the text fields. However, in the case of the name fields, we have 2 (or even more) text fields directly next to each other. Especially in the "Name Edit" view, where the current values are already entered, it helps visibility a lot to provide labels for each of the text fields.  
  To conclude my point of view, I would remove the bold formatting of the `NameFields` in `SpeziViews` for a consistent formatting but leave the rest as is. What is your point of view? Do you have a different point of view especially in regard to the labels?
3. We discussed disabling of buttons here https://github.com/StanfordSpezi/SpeziAccount/pull/7#discussion_r1312303932 already and came to the conclusion to disable them until all fields are valid.

Reply:
Thank you for the context!
Here are my suggestions:
1. I took a look at the settings app in iOS 17 and it seems like it models the exact same behaviour as you described. It has a "New" and "Verify" field for setting a new password. I would suggest to use the same wording and placeholder values and model all view like the Apple ID settings in the Settings App.
2.1. NameFields: That is a good point, I think having them non-bold sounds good. Regarding the long term plan: Does it make sense to move some views including these fields in a Spezi Account target that is independent of the account management and allows a reuse of the view components that are Account/User related? Might also include the user profile view that is currently part of the Views Package. Alternative would be to move them to the SpeziViews package, but I have the feeling that this is already crowded so it might make sense to rather push this here. What do you think?
2.2.: Regarding the labels: I looked at the Settings app and found a similar behaviour regarding the name fields, I would suggest to follow the same wording there. I would generally try to follow that example if reasonable to have that as a point of reference, same for the account creation process.
3. Agree, that sounds good!

Reply:
Thanks for the elaborating in such great detail on the action points.

1. We can move to introducing leading labels for the Password Change sheet.
2. (1) I created https://github.com/StanfordSpezi/SpeziViews/pull/16 to address the semibold fields labels. This doesn't require any changes in SpeziAccount and can be merged independently.
Any yes, it would definitely make sense to separate a lot of the infrastructure and views introduced in this PR into another target or SpeziViews. Maybe we could create a separate target over at SpeziViews depending on the relevancy?
(2) I will update the name field labels to be shorter, just as seen in the settings app.

Reply:
Sounds great, thank you!
Let's also create an issue that keeps track of the question where to move the view components. I could see that either adding a target here in Account or over in Views makes sense 👍

Reply:
Created an issue here https://github.com/StanfordSpezi/SpeziAccount/issues/19.

It might even make sense to create a completely new Spezi package for the Validation infrastructure. What do you think? Feel free to add your thoughts to the issue discussion.